corehq/apps/sso/utils/user_helpers.py

```diff
@@ -1,15 +1,18 @@
+import re
+
+
 def get_email_domain_from_username(username):
     """
     A quick utility for getting an Email Domain from a username
     (the last part of the email after the '@' sign)
     :param username: String
-    :return: Domain name string. `None` if there is no domain name.
+    :return: Domain name string. `None` if there is no domain name (or username is otherwise invalid).
     """
     split_username = username.split('@')
-    if len(split_username) < 2:
+    if len(split_username) != 2:
         # Not a real email address with an expected email domain.
-        return None
-    # we take the last split because `@` is technically an allowed character if
-    # inside a quoted string.
-    # See https://en.wikipedia.org/wiki/Email_address#Local-part
```

Comment:
Are we changing this to reject any email that does not have a single `@` character? According to the comment above (which is being removed) it is possible for an email address to have more than one.

Reply:
according to our security firm, we shouldn't expect emails to have that format

Reply:
In terms of the email address spec they are probably wrong, but practically speaking it might be fine.